lib/GLPI/Agent/Task/Inventory/Generic/Hostname.pm

```diff
@@ -21,9 +21,15 @@ sub doInventory {
     my (%params) = @_;
 
     my $inventory = $params{inventory};
-
-    # keep just the hostname
-    my $hostname = getHostname(short => 1);
+    my $assetname_support = $params{'assetname_support'};
+
+    # use the hostname as desired
+    my $hostname;
+    if ($assetname_support == 2 ) {
+	    $hostname = getHostname();
+    } else {
+	    $hostname = getHostname(short => 1);
+    }
```

Comment:
On lines 29 & 31, you mixed tabulation and spaces. This is why the linux test failed. Only spaces should be used.
```suggestion
    if ($assetname_support == 2 ) {
        $hostname = getHostname();
    } else {
        $hostname = getHostname(short => 1);
    }
```

Reply:
I updated the code, let's see the tests :-)
Maybe, future-wise, vi-rules could be added to the files?